src/Orleans.Runtime/MembershipService/MembershipTableManager.cs

```diff
@@ -273,14 +298,19 @@ public async Task UpdateStatus(SiloStatus status)
                 if (ok)
                 {
                     if (log.IsEnabled(LogLevel.Debug)) log.Debug("-Silo {0} Successfully updated my Status in the Membership table to {1}", myAddress, status);
+
                     var gossipTask = this.GossipToOthers(this.myAddress, status);
-                    if (status.IsTerminating())
+                    var timeoutTask = Task.Delay(GossipTimeout);
+                    var task = await Task.WhenAny(gossipTask, timeoutTask);
+                    if (ReferenceEquals(task, timeoutTask))
```

Comment:
Always wait up to 3 seconds for gossip to complete. May slow shutdown/startup, but also should give us generally more predictable behavior